OpenSim/Common/Component.cpp

```diff
@@ -1524,6 +1504,50 @@ void Component::dumpConnections() const {
     }
 }
 
+void Component::printSubcomponentInfo() const {
+    printSubcomponentInfo<Component>();
+}
+    
+template<typename C>
+void Component::printSubcomponentInfo() const {
```

Comment:
As a note, putting this definition in the cpp means that users will not be able to invoke the templated version. IMO it might be nice to expose the templated version, but I don't have a strong opinion. If we do expose the templated version, it would be great if the python or java tests could make sure the non-templated version works.